subsys/bluetooth/mesh/cdb.c

```diff
@@ -855,6 +867,11 @@ void bt_mesh_cdb_node_del(struct bt_mesh_cdb_node *node, bool store)
 
 	if (IS_ENABLED(CONFIG_BT_SETTINGS) && store) {
 		update_cdb_node_settings(node, false);
+
+		if (node->addr + node->num_elem > bt_mesh_cdb.lowest_avail_addr) {
```

Comment:
The in-memory increment should not be in the `if (IS_ENABLED(CONFIG_BT_SETTINGS) && store)` right?

Reply:
I think if the node is not removed from the persistent storage, then it will appear again after next reboot. If I'm not mistaken, the use case for `store` option is when a node was created but not stored in the persistent storage via `bt_mesh_cdb_node_store`.

Reply:
It will reappear on the next reboot, but in the meantime the lowest_avail_addr will cause the address to be available for reuse, no? Meaning
- Create node 0x200 without storing
- Node 0x200 does messaging and starts appearing in RPLs
- `bt_mesh_cdb_node_del(&node_200, false)` (or with persistence off)
- [Reassignment possible here]
- Reboot

Reply:
After the first step the node must be stored, otherwise it is an incorrect use of the API. Whenever you allocate a node that gets provisioned to the network -> store it. `bt_mesh_cdb_node_del` with `false` argument is called in 2 cases from what I see:
1. When was not restored from settings;
2. When provisioner didn't finish a node provisioning and was reset;

Reply:
Is it not possible to use the provisioner/CDB with `CONFIG_BT_SETTINGS=n`? Why all the conditionals on that throughout then?

Reply:
It is possible. Why to set `CONFIG_BT_SETTINGS=n` is a different question I think, but users can do this.

Reply:
And then they would be able to do this colission case I described above, right?

Reply:
Alright, I see your point.